core/math/expression.h

```diff
@@ -36,6 +36,12 @@
 class Expression : public Reference {
 	GDCLASS(Expression, Reference)
 public:
+	struct Context {
```

Comment:
This is added here and then unused, it should probably be added somewhere else.
Also, dont store a pointer, you should instead store an ObjectID via object->get_instance_id() and then resolve to pointer using ObjectDB::get_instance(). This way if the object was erased, it will fail and you can remove the watch.